File: src/models/country.py
import json

class Country:
    def __init__(self, alpha_2 : str) -> None:
        with open('./data/codigos_paises.json', 'r', encoding='utf-8') as file:
            paises = json.load(file)
            country = next((p for p in paises if p["alpha_2"] == alpha_2), None)
            self.code = country['code']
            self.alpha_2 = country['alpha_2']
            self.alpha_3 = country['alpha_3']
            self.name_pt = country['name_pt']
            self.name_en = country['name_en']
            
    def __str__(self):
        return "Country(alpha_2: " + self.alpha_2 + ")"
    
    def __repr__(self):
        return "Country(code: " + self.code + ", alpha_2: " + self.alpha_2 + ", alpha_3: " + self.alpha_3 + ", name_pt: " + self.name_pt + ", name_en: " + self.name_en + ")"
    
    def to_dict(self):
        return {
            "code": self.code,
            "alpha_2": self.alpha_2,
            "alpha_3": self.alpha_3,
            "name_pt": self.name_pt,
            "name_en": self.name_en
        }
    
    def from_dict(self, data : dict):
        self.code = data['code']
        self.alpha_2 = data['alpha_2']
        self.alpha_3 = data['alpha_3']
        self.name_pt = data['name_pt']
        self.name_en = data['name_en']
        
    def __eq__(self, other):
        return self.code == other.code
    
    def __hash__(self):
        return hash(self.code)
    

File: js/models/country.js
import fs from "fs";

class Country {
    constructor(alpha2) {
        const data = fs.readFileSync("./data/codigos_paises.json", "utf-8");
        const paises = JSON.parse(data);
        const country = paises.find(p => p.alpha_2 === alpha2);
        
        if (country) {
            this.code = country.code;
            this.alpha2 = country.alpha_2;
            this.alpha3 = country.alpha_3;
            this.namePt = country.name_pt;
            this.nameEn = country.name_en;
        } else {
            throw new Error(`Country with alpha_2 '${alpha2}' not found`);
        }
    }

    toString() {
        return `Country(alpha_2: ${this.alpha2})`;
    }

    toDict() {
        return {
            code: this.code,
            alpha_2: this.alpha2,
            alpha_3: this.alpha3,
            name_pt: this.namePt,
            name_en: this.nameEn
        };
    }

    fromDict(data) {
        this.code = data.code;
        this.alpha2 = data.alpha_2;
        this.alpha3 = data.alpha_3;
        this.namePt = data.name_pt;
        this.nameEn = data.name_en;
    }

    equals(other) {
        return this.code === other.code;
    }
}

export { Country };